File: src/main/slides-resources/jvmFunctional/code/for.scala
  def notFunErrors(events: List[Event], size: Int): List[Event] = {
    for {
      event <- events
      if event.isError
    } yield event
  }.take(size)

  def funErrors(events: List[Event], size: Int): List[Event] =
    events
      .filter(_.isError)
      .take(size)

File: src/main/slides-resources/jvmFunctional/code/for.kt
fun notFunErrors(events: List<Event>, size: Int): List<Event> {
    val result = mutableListOf<Event>()
    for (event in events) {
        if (event.isError) {
            result.add(event)
        }
        if (result.size >= size) {
            return result
        }
    }
    return result
}

fun funErrors(events: List<Event>, size: Int): List<Event> =
    events
        .filter { it.isError }
        .take(size)
